src/disc/stk/STKConnManager.hpp

```diff
@@ -67,6 +67,20 @@ class STKConnManager : public ConnManager {
   const GO * getConnectivity(LO localElmtId) const override {
     return &m_connectivity[m_elmtLidToConn[localElmtId]];
   }
+  
+  /** Get vector of bools associated to connectivity for a particular element, indicating whether the entity is owned by this rank
+    *
+    * \param[in] localElmtId Local element ID
+    *
+    * \returns vector of bools, with total size
+    *          equal to <code>getConnectivitySize(localElmtId)</code>
+    */
+  const std::vector<bool>& getOwnership(LO localElmtId) const override {
+     auto begin = m_ownership.begin()+m_elmtLidToConn[localElmtId];
+     auto end = begin+m_connSize[localElmtId];
+     m_owned.assign(begin,end);
```

Comment:
Why do you need `m_owned` on top of `m_ownership`?

Reply:
it's because vector<bool> is weird beast, as you said. Instead of returning a pointer to &m_ownership[i], as done for getConnectivity,  I'm creating a new subvector and returning it. In this way we are safe. If you prefer to convert it to vector<int> go for it.

Reply:
Yeah, I think the risk of small mods that cause unexpected behavior downstream (and hours of debugging) is not negligible. I wish they never implemented that specialization of std::vector... I will change it to something safer.